### ISSUE TITLE
Flicker when updating icon

### ISSUE THREAD

**Issue by Mathnerd314** [Loirooriol/tab-counter-plus#20]
When using SVG icons, and the tab is opened, there is often a flicker lasting a few frames where the old icon blinks out but the new icon is not drawn yet.

i.e. it goes something like [24] -> [&nbsp;&nbsp;&nbsp;] -> [25]

This doesn't happen as often when closing tabs, because (I think) the icon is cached and therefore Firefox just swaps out the bitmap rather than re-rendering the SVG. But if you open a lot of tabs and close them one-by-one the flicker does eventually return though.

Maybe there is some way to explicitly draw/cache the icon for +1 / -1 in the background so the flicker can be avoided? Obviously it would still flicker if you closed 10 tabs at once but in that case FF UI is slow enough anyway that it doesn't matter.

**Comment by Mathnerd314** [comment]
There's the background web page the addon runs in, maybe inserting a SVG there will render it.

Another idea would be rendering to canvas and using toDataUrl, PNG is probably faster.

**Comment by Mathnerd314** [comment]
Well, background doesn't work. I'll try canvas next.

**Comment by Mathnerd314** [comment]
Canvas doesn't help either.

I guess the flicker is really due to the indirection of using CSS variables for the image data, it has to load 6-7 data urls.

I couldn't find a Mozilla bug on file, but you filed https://bugzilla.mozilla.org/show_bug.cgi?id=1475877, which is close enough

**Comment by Loirooriol** [comment]
Firefox uses CSS variables so that it's easy to update the icon when the DPI changes. And in fact this is why I prefer SVG than canvas, I don't have to care about the size of the icon because it will be scaled as necessarily without becoming blurry.

**Comment by Mathnerd314** [comment]
Yeah, the only workable approach is to get the icons cached properly. The existing WebExtension caching code is borked, so maybe a decently powerful API for specifying a set of icons to all be cached w/ Firefox's internal cache could be approved...
